rosbag2_compression/src/rosbag2_compression/compression_factory_impl.hpp

```diff
@@ -20,67 +20,116 @@
 #include <stdexcept>
 #include <string>
 
+#include "pluginlib/class_loader.hpp"
+
 #include "logging.hpp"
 #include "rosbag2_compression/compression_factory.hpp"
-#include "rosbag2_compression/zstd_compressor.hpp"
-#include "rosbag2_compression/zstd_decompressor.hpp"
 
-namespace
+namespace rosbag2_compression
 {
 
-constexpr const char kCompressionFormatZstd[] = "zstd";
+using rosbag2_compression::BaseCompressorInterface;
+using rosbag2_compression::BaseDecompressorInterface;
+
+template<typename T>
+struct CompressionTraits
+{};
 
-/// Verify whether two case-insensitive chars are equal
-bool compare_char(const char c1, const char c2)
+template<>
+struct CompressionTraits<BaseCompressorInterface>
 {
-  return c1 == c2 || std::tolower(c1) == std::tolower(c2);
-}
+  static constexpr const char * name = "rosbag2_compression::BaseCompressorInterface";
```

Comment:
is there a difference between `constexpr const char * name` and `constexpr const char name[]`?

Reply:
yes - but I don't understand what it is. It builds as `char * name`, but not as `char name[]`, like so

```
/usr/bin/ld: librosbag2_compression.so: undefined reference to `rosbag2_compression::CompressionTraits<rosbag2_compression::BaseCompressorInterface>::name'
```